backend/src/server.ts

```
@@ -1,21 +1,87 @@
 import { config } from "dotenv";
+import { Request, Response, NextFunction } from 'express';
 import { loadDatabaseDriver } from "./repo/Driver";
+import { loadTransporter } from "./service/auth";
 import { User, UserRegistration, UserRepository } from "./domain/User";
 
 const express = require("express");
 const bodyParser = require("body-parser");
 const cors = require("cors");
 const path = require("path");
+const crypto = require("crypto"); // generate token
 
 const app = express();
 app.use(cors());
 app.use(bodyParser.json());
+app.use(express.json());
 
 config();
 
 let driver = loadDatabaseDriver();
+let transporter = loadTransporter();
 
-app.post("/api/login", async (req, res, next) => {
+// In-memory store for demo purposes (replace with DB or Redis in production)
+const tokenStore = {};
+
+// Route to accept email and send verification
+app.post("/api/register", async (req: Request, res: Response) => {
+  
+  const { email } = req.body;
+  const db = driver;
+
+  // Check for email, then see if one ties to a user
+  if (!email) return res.status(400).json({ error: "Email is required" });
+
+  // make sure there's not already a registered user with this login
+  const existingUser = await db.userRepository.GetByEmail(email);
+  if (existingUser != null) {
+    return res.status(400).json({ error: "User already exists!" });
```

Silently fail, and tell the user that the verification code was sent to not reveal that a user exists with this email.